ProjectSidewalk/SidewalkWebpage
# 324: Showing completed streets as part of current mission after reloading mission

Resolves #324
Fixes #463 (after edits by @misaugstad)

If you were to reload a mission at least once while working on it, streets that were audited before reloading would show up as streets completed in previous missions on the mission complete map. To fix this, I added a column to AuditTaskTable that keeps track of the mission id that it is currently apart of along with functions that help obtain tasks that were part of a certain mission. Now, the code in Mission.js checks gets tasks that are part of the mission (if applicable) when creating (or recreating) a mission after loading.

Testing instructions:
1. Starting as not signed in, go through multiple missions (no need to actually label yet, just "walk through" them). Test multiple version of missions where you:
- Don't reload at all
- Get stuck/move locations
- Reload once at an intersection/turning point
- Reload 2 or more times at an intersection/turning point (you can also do anywhere, but an intersection/turning point is more likely to be an end of a street)
2. Now sign in and test the same versions (don't need to be as thorough, this is just to make sure that it works the same way when signed in)
3. Also, test that labelling still works by labelling a few things

Things to look out for:
- Streets completed in current mission that are not colored green
- Streets/other areas that are colored green when not supposed to
- Random "ghost" green spots
- Labelling function not working/being counted correctly at the end
